app/questionnaire/value_source_resolver.py

```diff
@@ -230,20 +230,20 @@ def _resolve_list_value_source(self, value_source: Mapping) -> int | str | list:
 
         return list(list_model)
 
-    def _resolve_calculated_summary_value_source(
+    def _resolve_summary_with_calculation(
         self, value_source: Mapping, *, assess_routing_path: bool
     ) -> IntOrDecimal | None:
-        """Calculates the value for the 'calculation' used by the provided Calculated Summary.
+        """Calculates the value for the 'calculation' used by the provided Calculated or Grand Calculated Summary.
 
-        The caller is responsible for ensuring the provided Calculated Summary and its answers are on the path,
+        The caller is responsible for ensuring the provided Summary and its components are on the path
         or providing routing_path_block_ids when initialising the value source resolver.
         """
-        calculated_summary_block: ImmutableDict = self.schema.get_block(value_source["identifier"])  # type: ignore
-
-        if not self._is_block_on_path(calculated_summary_block["id"]):
+        summary_block: ImmutableDict = self.schema.get_block(value_source["identifier"])  # type: ignore
+        if not self._is_block_on_path(summary_block["id"]):
             return None
 
-        calculation = calculated_summary_block["calculation"]
+        calculation = summary_block["calculation"]
```

Comment:
This confused me a little bit at first. Can you change it to something like `# old calculated summary block's calculation object may have answers_to_calculate instead of operation`? Or am I getting it wrong?

Reply:
Yep good spot, it should be operation instead of calculation, have updated the comment so it hopefully is a bit clearer?

Reply:
Yeah, thanks. It wasn't some big issue, non-blocking change really.